packagekit: Improve visuals of updates-history and flow list

garrett opened the pull request:
Depends on #14581

garrett commented:
I fixed some layout issues on the software page, but the CSS depends on changes in the aforementioned issue, so I'm sending this as a draft PR to be included as a follow-up, instead of detangling the CSS, which would just cause merge conflicts with the other PR.

Rebasing on master once the other PR lands should drop the other two commits.

garrett commented:
Highlights of the (relatively minor) changes:
- Moved icon next to date
- Adjusted padding around the flowlist
- Changed font weight of the flowlist to be normal (not bold)

![Screen Shot 2020-09-14 at 12 54 52](https://user-images.githubusercontent.com/10246/93077736-8a54ff00-f689-11ea-993b-e7a5e40eab88.png)

skobyda commented:
Looks good

skobyda commented:
I checked it out locally and I see you are changing stuff related to update history. I don't really touch that in my PR, so you can safely drop out my commits from this PR and merge your CSS fixes independently.

garrett commented:
> I don't really touch that in my PR, so you can safely drop out my commits from this PR and merge your CSS fixes independently.

When I rebased on master, I had merge conflicts in the CSS. Hence, making this a follow-up, instead of detangling it myself, then forcing you to detangle it too.

garrett commented:
The flow should be something like this (text is imprecise and this is a very low-fi rough wireframe design-wise):

![flow: software update: system-needs-reboot](https://raw.githubusercontent.com/cockpit-project/cockpit-design/master/software-updates/flow-needs-reboot.png)

Some of this, such as the reboot button being a split dropdown like the system overview, could be deferred to another PR. But this PR should have much of the general flow in the wireframe.

Important parts:

1. **Let people know what's up as soon as possible**. They should expect that their system may need to be rebooted and that services might need to be restarted. We _should_ have a quick and dirty check that gets this mostly correct, but might be wrong. Hence _may_. Having it all the time  (whenever updates are available) would be bad, however, as people will just ignore it. So it should be based on some sort of heuristics.
2. Show the updates as they happen. (We have this already, even in master.)
3. **Provide feedback when the system is doing something**, such as checking to see if services need restarting and/or the system needs rebooting. We currently do not do this in the PR. It's absolutely mandatory to have this step.
4. Have **three states** at the end: 
   1. **Updated** (and doesn't need intervention).
   2. **Services need restarting**. (Simple list of services that need to be restarted, with all checked by default.)
   3. **System _needs_ a reboot — with a button to reboot the machine**.
      - Ideally with a split button like overview. But that can be a future PR. 
      - Minimally, the action needs to exist, so a simple button should be enough.
      - We could do the reboot / ignore pattern, but switch ignore to manage services, which would then show the services need restarting dialog. But we'd need to keep the system needs rebooting state with an action button, perhaps in a banner or alert in the software page _and_ in the overview. Or we could make it a top-level shell thing (in another PR).

I've bolded things which do not currently exist in the PR, at least from what I saw earlier today.

Hopefully this helps provide a direction. Thanks for doing the groundwork of integration! I apologize for not putting this all together in a form like this already (I've been busy with other stuff).

martinpitt commented:
@garrett: Is there still somethign left over after PR #15089 ?

garrett commented on pkg/packagekit/updates.scss:
```diff
@@ -362,6 +361,19 @@ table.header-buttons {
     }
 }
 
-.pk-updates .pf-c-description-list + div {
```
pk-updates no longer exists; removed this as a result

garrett commented:
I've detangled this PR and rebased; it should be ready for review.

It mainly just adjusts the history so the vertial alignment of the text and icons in the list is correct and the packages in the list have proper weight (default instead of bold).

KKoukiou commented on pkg/packagekit/updates.scss:
```diff
@@ -271,8 +271,8 @@ div.changelog {
 }
 
 .flow-list {
+    margin: 0 1rem 1rem;
```
flow list is direct child of pf-c-table__expandable-row-content. So by adding left and right margin there, you basically add to pf-c-table__expandable-row-content. So if the expandable rows needs margins, that change should be in cockpit-components-listing-panel.scss and not specifically here. What do you think?

KKoukiou replied:
I took a look again, this margin is wrong. It creates a difference spacing for the expanded content for this list compared to the list above, and since there is already margin on td this is not needed.

If this is removed I am fine with the rest of the changes.

KKoukiou reviewed (changes requested):
One comment, other than that I am good with it. Please rebase to master if you will repost so that we can test it fast with make_dist :)

garrett commented:
Rebased, but haven't addressed https://github.com/cockpit-project/cockpit/pull/14616#discussion_r579005013.

KKoukiou commented:
> Rebased, but haven't addressed [#14616 (comment)](https://github.com/cockpit-project/cockpit/pull/14616#discussion_r579005013).


heh, what's the point :P ? Shall I wait that you answer/address the comment before reviewing again?

garrett commented:
> what's the point

I wanted to make sure the PR didn't languish longer to introduce more conflicts. (There were conflicts.) When I finished rebasing, it was time for lunch.

garrett commented:
Rebased, fixed conflicts, dropped margin, and ensured the content body lines up with the content body of the other expanded lists on the page (by removing the flow-list li padding).

FWIW: I think we could implement flow-list in a better way with much less code using display: flex these days. Perhaps we could even use PF4's flex. Either is fine for a follow-up though, I think?

KKoukiou reviewed (approved):
I am generally not happy about PF4 reoverriding the CSS included in this PR in mobile screen sizes, but if you @garrett think is fine feel free to merge it.

martinpitt commented:
Removing obsolete "needswork" label, I think this got fixed. Sorry @garrett , I think this fell through the cracks. I force-pushed a rebase, and will let the tests run again.
